tests/foreman/ui_airgun/test_contentcredential.py

```diff
@@ -504,3 +500,245 @@ def test_positive_update_key_for_repo_from_product_with_repos(
+        """Create gpg key with valid name and valid gpg key then
+        associate it to repository from custom product that has one repository
+        then delete the key
+
+        :id: 92ba492e-79af-48fe-84cb-763102b42fa7
+
+        :expectedresults: gpg key is associated with single repository during
+            creation but removed from repository after deletion
+
+        :CaseLevel: Integration
+        """
+        gpg_key = entities.GPGKey(
+            content=self.key_content,
+            organization=self.organization,
+        ).create()
+        # Creates new product without selecting GPGkey
+        product = entities.Product(
+            name=gen_string('alpha'),
+            organization=self.organization,
+        ).create()
+        # Creates new repository with GPGKey
+        repo = entities.Repository(
+            name=gen_string('alpha'),
+            url=FAKE_1_YUM_REPO,
+            product=product,
+            gpg_key=gpg_key,
+        ).create()
+        with session:
+            session.organization.select(org_name=self.organization.name)
+            # Assert that GPGKey is associated with product
+            values = session.contentcredential.read(gpg_key.name)
+            assert len(values['products']['table']) == 0
+            assert len(values['repositories']['table']) == 1
+            assert values['repositories']['table'][0]['Name'] == repo.name
+            repo_values = session.repository.read(product.name, repo.name)
+            assert repo_values['repo_content']['gpg_key'] == gpg_key.name
+            session.contentcredential.delete(gpg_key.name)
+            # Assert GPGKey isn't associated with repository
+            repo_values = session.repository.read(product.name, repo.name)
+            assert not repo_values['repo_content']['gpg_key']
+
+    @tier2
+    @upgrade
+    def test_positive_delete_key_for_repo_from_product_with_repos(
+            self, session):
+        """Create gpg key with valid name and valid gpg key then
+        associate it to repository from custom product that has more than
+        one repository then delete the key
+
+        :id: 5f204a44-bf7b-4a9c-9974-b701e0d38860
+
+        :expectedresults: gpg key is associated with single repository but not
+            with product during creation but removed from repository after
+            deletion
+
+        :BZ: 1461804
+
+        :CaseLevel: Integration
+        """
+        # Creates New GPGKey
+        gpg_key = entities.GPGKey(
+            content=self.key_content,
+            organization=self.organization,
+        ).create()
+        # Creates new product without GPGKey association
+        product = entities.Product(
+            name=gen_string('alpha'),
+            organization=self.organization,
+        ).create()
+        # Creates new repository_1 with GPGKey association
+        repo1 = entities.Repository(
+            gpg_key=gpg_key,
+            name=gen_string('alpha'),
```

Comment:
i'd drop this redundant line from every entity of ported tests :) but that's definitely low-priority nitpick